### DISCUSSION TITLE
xy-pad vector audio mixer

### DISCUSSION THREAD

**Discussion by inst5** [cables-gl/cables_docs#187] [Answers & Questions] [answered]
Hi,
Is it possible to mix 4 audio files with a Ops.Sidebar.XYPad ?
So in each corner you can listen to 100% of a single sound file and in the midlle of the xy Pad you can hear them all.

basicly something like this:  
https://www.kvraudio.com/product/sanford_vector_pad_by_sanford_sound_design      
But with audio files.

thanks for feeeback.

**Comment by steam0r** [comment, marked as answer]
hey,

take a look at this op and the examples: https://cables.gl/op/Ops.WebAudio.Mixer

xypad produces x/y from 0-1 and that might be compatible with the gain of the mixer, i assume.

**Reply by inst5** [reply]
Hi,
thanks! 
Yes tried it out with an xy-pad controller op.
So basicly it´s that but 2 chanels only. - So I will try it with 4 channels too.

**Comment by pandrr** [comment]
maybe you just want to remap the xy values?

like this:
https://cables.gl/edit/pHchzZ

**Comment by inst5** [comment]
Actually the behavoir in your patch is a bit different. Always 2 sources are audible.
I meant an xy pad like this: https://books.google.at/books?id=EDkPEAAAQBAJ&pg=PA144&lpg=PA144&dq=xy+pad+with+crossfades&source=bl&ots=sX2Rm5y-Km&sig=ACfU3U2Jm9ViC-BzupagDW6OXPzoQGJwHw&hl=de&sa=X&ved=2ahUKEwiAuNv_w7P2AhWqSPEDHTbrBS4Q6AF6BAgOEAM#v=onepage&q=xy%20pad%20with%20crossfades&f=false

But it works with taking 2 calbes crossfades op. by taking the xy-number of the xy-pad to 2 crossfades, multiplying the values of the output and goig into 4 seperate gains of 4 audio sources

**Reply by pandrr** [reply]
for this you can just tweak the values in the maprange ops to meet your wishes

**Reply by inst5** [reply]
ok, will try that as option B.
This would be also interesting for not only considering xy-axis but also taking the z-axis as mixing value.
